python/python/bystro/ancestry/train.py

```diff
@@ -308,6 +313,88 @@ def _calc_fst(variant_counts: pd.Series, samples: pd.DataFrame) -> float:
     return (p * (1 - p) - total) / (p * (1 - p))
 
 
+def load_1kgp_vcf(vcf_filepath: str) -> pd.DataFrame:
+    """Temporary placeholder method to load in 1kgp vcf
+    filtered down to the same variants as the gnomad loadings for testing
+    using plink2.
+    """
+    #This line will have to be altered if testing other vcfs
+    header_vcf=pd.read_csv(vcf_filepath,delimiter='\t',skiprows=107)
+    dosage_vcf = header_vcf.replace("0|0", 0)
+    dosage_vcf = dosage_vcf.replace("0|1", 1)
+    dosage_vcf = dosage_vcf.replace("1|0", 1)
+    dosage_vcf = dosage_vcf.replace("1|1", 2)
+    #If testing other vcfs this line will also need to be altered
+    dosage_vcf=dosage_vcf.rename(
+        columns={"#CHROM": "Chromosome", "POS": "Position"}, 
+        errors="raise"
+    )
+    return dosage_vcf
+    
+
+def load_pca_loadings(GNOMAD_PC_PATH: str, dosage_vcf: pd.DataFrame) -> pd.DataFrame:
+    #Gnomad pc file 
+    """Load in the gnomad PCs and reformat for PC transformation.
+    """
+    loadings = pd.read_csv(GNOMAD_PC_PATH, sep="\t")
+    #Gnomad pc loadings file includes additional formatting that needs to be sanitized 
+    loadings[["Chromosome", "Position"]] = loadings["locus"].str.split(":", expand=True)
+    #Match variant format of gnomad loadings with 1kgp vcf
+    get_chr_pos = lambda x: x["locus"][3:]
+    get_ref_allele = lambda x: x["alleles"][2]
+    get_alt_allele = lambda x: x["alleles"][6]
+    loadings["variant"] = loadings.apply(
+        lambda x: ":".join([get_chr_pos(x), get_ref_allele(x), get_alt_allele(x)]), axis=1
+    )
+    #Remove brackets
+    loadings["loadings"] = loadings["loadings"].str[1:-1]
+    #Split PCs and join back
+    gnomadPCs = loadings["loadings"].str.split(",", expand=True)
+    loadings = loadings.join(gnomadPCs)
+    loadings = loadings.reset_index()
+    pc_range = range(8, 38)
+    pc_loadings = loadings.iloc[:, pc_range].copy()
+    pc_loadings["variant"] = loadings["variant"]
+    pc_loadings = pc_loadings.set_index("variant")
+    pc_loadings = pc_loadings.sort_index()
+    return pc_loadings
+
+
+def loadings_var_overlap(pc_loadings: pd.DataFrame, dosage_vcf: pd.DataFrame) -> tuple[pd.DataFrame, pd.DataFrame]:
```

Comment:
if we defined dosage_vcf so that its index is sample_ids and columns are variants, we could do something like the following here:
```
def loadings_var_overlap(pc_loadings: pd.DataFrame, dosage_vcf: pd.DataFrame) -> tuple[pd.DataFrame, pd.DataFrame]:
    common_variants = pc_loadings.index.intersection(dosage_vcf.columns)
    return pc_loadings.loc[common_variants]
```

wdyt?

Reply:
The current format of dosage_vcf has variant IDs as rows and sample IDs in the header/columns, with additional informational columns. It's not until genos is processed and transposed that its index is sample IDs and columns are variant IDs. If I move genos processing to a separate function above this step (as you suggested below) that would make sense to do a genos_transpose columns/pc_loadings index intersection or maybe even easier conceptually genos/pc_loadings index intersection. Worth enforcing genos[overlapping_variants] as well in some step (maybe this one) since in the current version the dosage_vcf is pre-processed filtered on the gnomad variants.